src/entities/mellow/mellowLpRouter.ts

```diff
@@ -633,6 +633,61 @@ class MellowLpRouter {
       throw new Error('Unsucessful rollover confirmation.');
     }
   };
+
+  registerForAutoRollover = async (registration: boolean): Promise<number> => {
+    if (
+      isUndefined(this.readOnlyContracts) ||
+      isUndefined(this.writeContracts) ||
+      isUndefined(this.userAddress)
+    ) {
+      throw new Error('Uninitialized contracts.');
+    }
+
+    try {
+      await this.writeContracts.mellowRouter.callStatic.registerForAutoRollover(registration);
+    } catch (err) {
+      console.error('Error during registration for auto-rollover', err);
+      throw new Error('Unsuccessful auto-rollover registration simulation');
+    }
+
+    const gasLimit = await this.writeContracts.mellowRouter.estimateGas.registerForAutoRollover(
+      registration,
+    );
+
+    const tx = await this.writeContracts.mellowRouter.registerForAutoRollover(registration, {
+      gasLimti: getGasBuffer(gasLimit),
+    });
+
+    try {
+      const receipt = await tx.wait();
+
+      try {
+        await this.refreshWalletBalance();
```

Comment:
We don't need to refresh wallet balance after registration.

Reply:
resolved